hypothesis-python/src/hypothesis/searchstrategy/types.py

```diff
@@ -61,9 +62,19 @@ def try_issubclass(thing, superclass):
         return False
 
 
+def is_a_new_type(thing):
+    """Returns True if thing is a NewType."""
+    return (
+                hasattr(thing, "__supertype__")
+                and hasattr(typing, "NewType")
+                and inspect.isfunction(thing)
+                and getattr(thing, "__module__", 0) == "typing"
```

Comment:
Probably my fault that it was like this, but `getattr(thing, "__module__", None)` would be a nicer idiom.